languageFactories/cwl-v1-0/src/main/scala/languages/cwl/CwlV1_0LanguageFactory.scala

```diff
@@ -28,15 +30,25 @@ class CwlV1_0LanguageFactory(override val config: Config) extends LanguageFactor
                                  workflowIdForLogging: WorkflowId,
                                  ioFunctions: IoFunctionSet,
                                  importResolvers: List[ImportResolver]): Parse[ValidatedWomNamespace] = {
+
+    def parse(): Parse[Cwl] = source.workflowUrl match {
+      case Some(url) => for {
+        reference <- fromEither[IO](CwlReference.fromString(url).map(Right(_)).getOrElse(Left(NonEmptyList.one(s"Invalid workflow reference: $url")))): Parse[CwlReference]
```

Comment:
It seems like somewhere in all this abstraction we must be making an HTTP requests to actually get the `url`. How is it doing that?

Reply:
Eventually all this goes into a call to `cwltool`, which accepts commands in the form of `cwltool --print-pre URL` and gives us back an appropriate CWL string

Reply:
Ah OK, I was hunting around in our Scala code for this but it makes total sense that we're handing it off